commit db014de5cd1ede1d0570a987781748e32bbb01ca (refs/changes/39/4539/1)
Author: Robert Bogart <robert.bogart@joyent.com>
Date:   2018-07-18T22:56:17+00:00 (1 year, 3 months ago)
    
    MORAY-480 Moray should pass an artedi collector to Cueball to expose connection-related failure events

diff --git a/README.md b/README.md
index 96e6810..78a1cc2 100644
--- a/README.md
+++ b/README.md
@@ -243,6 +243,7 @@ The following metrics are always collected:
 - Backend request queue length
 - End-to-end latency for all requests
 - Count of requests completed
+- Count of of all Cueball failure events
 
 Each of the metrics returned include the following metadata labels:
 
diff --git a/lib/pg.js b/lib/pg.js
index 02b25cc..c1d6949 100644
--- a/lib/pg.js
+++ b/lib/pg.js
@@ -752,7 +752,8 @@ function PGPool(options) {
                 delay: 250,
                 maxDelay: 2500
             }
-        }
+        },
+        collector: options.collector
     });
 
     this.url = options.url;
diff --git a/package.json b/package.json
index ef079bb..edebab2 100644
--- a/package.json
+++ b/package.json
@@ -13,7 +13,7 @@
         "bunyan-syslog": "0.2.2",
         "clone": "0.1.11",
         "crc": "0.2.1",
-        "cueball": "2.5.2",
+        "cueball": "2.7.0",
         "dtrace-provider": "~0.8",
         "deep-equal": "0.0.0",
         "fast": "2.3.2",
